kore/src/Kore/Log/WarnRestartSolver.hs

```diff
@@ -0,0 +1,45 @@
+{- |
+Copyright   : (c) Runtime Verification, 2022
+License     : BSD-3-Clause
+-}
+module Kore.Log.WarnRestartSolver (
+    WarnRestartSolver,
+    warnRestartSolver,
+) where
+
+import qualified Control.Monad.Catch as Exception
+import Log
+import Prelude.Kore
+import Pretty (
+    Pretty (..),
+ )
+import qualified Pretty
+import SMT.SimpleSMT (SolverException)
+
+newtype WarnRestartSolver = WarnRestartSolver
+    {solverException :: SolverException}
+    deriving stock (Show)
+
+instance Pretty WarnRestartSolver where
+    pretty WarnRestartSolver{solverException} =
+        Pretty.vsep
+            [ "The SMT solver crashed with the following exception:"
+            , Pretty.indent 4 (pretty $ Exception.displayException solverException)
+            , "Will restart and reinitialise the solver\
```

Comment:
Is this similar to other log messages? "Will restart ..." reads a bit strangely; more natural would be "I will restart ..." or "Restarting and reinitialising ..." If this is similar to other messages though, it should stay.

Reply:
Done, thanks!